create/cloudvm.go

```diff
@@ -0,0 +1,141 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	runtimev1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	infrastructure "github.com/ninech/apis/infrastructure/v1alpha1"
+	meta "github.com/ninech/apis/meta/v1alpha1"
+	"github.com/ninech/nctl/api"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type cloudVMCmd struct {
+	Name                string            `arg:"" default:"" help:"Name of the CloudVM instance. A random name is generated if omitted."`
+	Location            string            `default:"nine-es34" help:"Location where the CloudVM instance is created."`
+	MachineType         string            `default:"" help:"MachineType defines the sizing for a particular cloud vm."`
+	Hostname            string            `default:"" help:"Hostname allows to set the hostname explicitly. If unset, the name of the resource will be used as the hostname. This does not affect the DNS name."`
+	PowerState          string            `default:"on" help:"PowerState specifies the power state of the cloud VM. A value of On turns the VM on, shutdown sends an ACPI signal to the VM to perform a clean shutdown and off forces the power off immediately."`
+	OS                  string            `default:"" help:"OS which should be used to boot the VM."`
+	BootDiskSize        string            `default:"20Gi" help:"BootDiskSize that will be used to boot the VM from."`
+	Disks               map[string]string `default:"{}" help:"Disks specifies which additional disks to mount to the machine."`
+	PublicKeys          []string          `default:"[]" help:"PublicKeys specifies the SSH Public Keys that can be used to connect to the VM as root. The keys are expected to be in SSH format as defined in RFC4253. Immutable after creation."`
+	PublicKeysFromFile  []string          `short:"public-keys-f" predictor:"file" help:"CloudConfig via file. Has precedence over args. PublicKeys specifies the SSH Public Keys that can be used to connect to the VM as root. The keys are expected to be in SSH format as defined in RFC4253. Immutable after creation."`
+	CloudConfig         string            `default:"" help:"CloudConfig allows to pass custom cloud config data (https://cloudinit.readthedocs.io/en/latest/topics/format.html#cloud-config-data) to the cloud VM. If a CloudConfig is passed, the PublicKey parameter is ignored. Immutable after creation."`
+	CloudConfigFromFile string            `short:"cloud-config-f" predictor:"file" help:"CloudConfig via file. Has precedence over args. CloudConfig allows to pass custom cloud config data (https://cloudinit.readthedocs.io/en/latest/topics/format.html#cloud-config-data) to the cloud VM. If a CloudConfig is passed, the PublicKey parameter is ignored. Immutable after creation."`
```

Comment:
I don't think you can use short names with dashes in them: `go run . --help
panic: cloudVMCmd.PublicKeysFromFile: invalid short flag name "public-keys-f": invalid rune`

Reply:
fixed, changed them both to not have short names